src/containerapp/azext_containerapp/custom.py

```diff
@@ -2180,91 +2180,108 @@ def setup_core_dns(cmd, distro=None, kube_config=None, kube_context=None, skip_s
         raise ValidationError(error)
 
     original_folder, folder_status, error = create_sub_folder(parent_folder, "original")
+    if not folder_status:
+        raise ValidationError(error)
+    new_filepath_with_timestamp, folder_status, error = create_sub_folder(parent_folder, "new")
     if not folder_status:
         raise ValidationError(error)
 
     kube_client = create_kube_client(kube_config, kube_context, skip_ssl_verification)
 
-    # backup original deployment and configmap
-    logger.info("Backup existing coredns deployment and configmap")
-    original_coredns_deployment = get_core_dns_deployment(kube_client, original_folder)
-    coredns_deployment = copy.deepcopy(original_coredns_deployment)
-
-    original_coredns_configmap = get_core_dns_configmap(kube_client, original_folder)
-    coredns_configmap = copy.deepcopy(original_coredns_configmap)
-
-    volumes = coredns_deployment.spec.template.spec.volumes
-    if volumes is None:
-        raise ValidationError('Unexpected Volumes in coredns deployment, Volumes not found')
-
-    volume_mounts = coredns_deployment.spec.template.spec.containers[0].volume_mounts
-    if volume_mounts is None:
-        raise ValidationError('Unexpected Volume mounts in coredns deployment, VolumeMounts not found')
-
-    coredns_configmap_volume_set = False
-    custom_coredns_configmap_volume_set = False
-    custom_coredns_configmap_volume_mounted = False
-
-    for volume in volumes:
-        if volume.config_map is not None:
-            if volume.config_map.name == CORE_DNS:
-                for mount in volume_mounts:
-                    if mount.name is not None and mount.name == volume.name:
-                        coredns_configmap_volume_set = True
-                        break
-            elif volume.config_map.name == CUSTOM_CORE_DNS:
-                custom_coredns_configmap_volume_set = True
-                for mount in volume_mounts:
-                    if mount.name is not None and mount.name == volume.name:
-                        custom_coredns_configmap_volume_mounted = True
-                        break
-
-    if not coredns_configmap_volume_set:
-        raise ValidationError("Cannot find volume and volume mounts for core dns config map")
-
-    original_custom_core_dns_configmap = backup_custom_core_dns_configmap(kube_client, original_folder)
+    if distro == AKS_AZURE_LOCAL_DISTRO or distro is None:
```

Comment:
Not quite sure why I set distro default to None, but actually we should not allow none. This command should specify distro explicitly